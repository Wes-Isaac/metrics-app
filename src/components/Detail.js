import React from 'react';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { FaChevronCircleLeft } from 'react-icons/fa';
import Info from './Info';

export default function Detail() {
  const detail = useSelector((state) => state.detail);
  let name;
  let info;
  if (detail[0]) {
    name = detail[0].name;
    info = detail[0].info;
  }

  return (
    <div>
      <div className="detail-header">
        <Link to="/"><FaChevronCircleLeft className="sign" /></Link>
        <h2>{detail[0] && name }</h2>
      </div>
      {info && <Info info={info} />}
    </div>
  );
}
