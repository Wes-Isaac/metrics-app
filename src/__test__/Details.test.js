import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import Detail from '../components/Detail';

describe('Render components', () => {
  test('render DetailPage', () => {
    const tree = renderer
      .create(<Router><Provider store={store}><Detail /></Provider></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
