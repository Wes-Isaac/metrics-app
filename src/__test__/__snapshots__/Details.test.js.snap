// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render components render DetailPage 1`] = `
<div>
  <div
    className="detail-header"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <svg
        className="sign"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 504C119 504 8 393 8 256S119 8 256 8s248 111 248 248-111 248-248 248zM142.1 273l135.5 135.5c9.4 9.4 24.6 9.4 33.9 0l17-17c9.4-9.4 9.4-24.6 0-33.9L226.9 256l101.6-101.6c9.4-9.4 9.4-24.6 0-33.9l-17-17c-9.4-9.4-24.6-9.4-33.9 0L142.1 239c-9.4 9.4-9.4 24.6 0 34z"
        />
      </svg>
    </a>
    <h2 />
  </div>
</div>
`;
